src/System/Console/Hawk/Lock.hs

```diff
@@ -20,106 +20,32 @@ module System.Console.Hawk.Lock
 
+-- | Create a temporary file to indicate to other hawk process
+--  not to run until this process is unblocked
 withLock :: IO a -> IO a
-withLock = withSocketLock False
+withLock action = withFileLock lockfile Exclusive (const action) >>= cleanup
 
+-- | withLock but with additional messages
 withTestLock :: IO a -> IO a
-withTestLock = withSocketLock True
-
-
-type Lock = Socket
-
-lock :: Bool -> IO Lock
-lock testing = catchJust isADDRINUSE openSocket $ \() -> do
-    -- open failed, the lock must be in use.
-    
-    when testing $ putStrLn "** LOCKED **"
-    
-    -- used to test an interleaving in which the socket is closed here,
-    -- between openSocket and waitForException.
-    when testing $ threadDelay 20000
-    
-    -- wait for the other instance to signal that it is done with the lock.
-    catchJust isDisconnected waitForException $ \_ -> do
-      -- we were disconnected, the server must have released the lock!
-      
-      when testing $ printf "** UNLOCKED **\n"
-      
-      -- try again.
-      lock testing
-
-unlock :: Lock -> IO ()
-unlock = closeSocket
-
-
-waitForException :: IO a
-waitForException = bracket openHandle closeHandle $ \h -> do
-    s <- hGetContents h
-    length s `seq` return ()  -- blocks until EOF, which never comes
-                              -- because the server never accepted the connection
-    error $ printf "port %s in use by a program other than hawk" $ show portNumber
-
-
-isADDRINUSE :: IOError -> Maybe ()
-isADDRINUSE = guard . (== "bind") . ioe_location
-
-isDisconnected :: IOError -> Maybe String
-isDisconnected = fmap (xs !!) . indexOf . ioe_location
-  where
-    xs = ["connect", "hGetContents"]
-    indexOf x = elemIndex x xs
-
-
-openHandle :: IO Handle
-openHandle = connectTo "localhost" $ PortNumber portNumber
-
-closeHandle :: Handle -> IO ()
-closeHandle = hClose
-
-
-openSocket :: IO Socket
-openSocket = listenOn portNumber
-
-closeSocket :: Socket -> IO ()
-closeSocket = sClose
-
-
--- the first few [0-9] characters of the sha1 of "hawk"
-portNumber :: PortNumber
-portNumber = 62243
-
--- from the source of Network.listenTo
-listenOn :: PortNumber -> IO Socket
-listenOn port = do
-    proto <- getProtocolNumber "tcp"
-    bracketOnError
-        (socket AF_INET Stream proto)
-        (sClose)
-        (\sock -> do
-            -- unlike the original listenOn, we do NOT set ReuseAddr.
-            -- this way the call will fail if another instance holds the lock.
-            --setSocketOption sock ReuseAddr 1
-            bindSocket sock (SockAddrInet port iNADDR_ANY)
-            listen sock maxListenQueue
-            return sock
-        )
+withTestLock action = do
+    result <- withFileLock lockfile Exclusive $ \_ -> do
+        putStrLn "** LOCKED **"
+        res <- action
+        putStrLn "** UNLOCKED **"
+        return res
+    cleanup result
+
+-- A temporary file - to be deleted after unlock
+lockfile :: FilePath
+lockfile = "lock"
```

Comment:
I'm guessing this will create a file in the current folder? That sounds like a bad idea, who knows if the user already has a file they care about called "lock" in that folder. How about creating it in hawk's context directory?